src/queryCache.js

```diff
@@ -110,7 +110,7 @@ export function makeQueryCache() {
 
     if (query) {
       Object.assign(query, { queryVariables, queryFn })
-      Object.assign(query.config, config)
+      query.config = { ...query.config, ...config }
```

Comment:
@tannerlinsley 

`Object.assign` was causing a bug. Consider this situation:
1. first instance of useQuery creates a new query and sets query.config pointer to config object
2. second instance of useQuery with different config object takes query from cache
3. `Object.assign` mutates first instance `query.config` object

Now as I think about it, maybe we should copy query config in the first step instead of referencing it?
Either way, it should be good now.

But that was super tough to debug :D